bundle/src/main/java/com/adobe/acs/commons/wcm/impl/PublishUrlServlet.java

```diff
@@ -0,0 +1,116 @@
+/*-
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2013 - 2024 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.wcm.impl;
+
+import com.day.cq.commons.Externalizer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.servlets.annotations.SlingServletResourceTypes;
+import org.jetbrains.annotations.NotNull;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+import javax.servlet.Servlet;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/**
+ * This servlet generates a JSON response with the externalized URLs for the given path using configured keys in its
+ * configuration. They keys need to match the environment keys configured in the Externalizer configuration. The servlet
+ * uses Externalizer service to generate the externalized URLs via Externalizer.externalLink() method. The response is
+ * in format {"Author": "Author URL", "Publish": "Publish URL", ...}.
+ */
+@Component(service = Servlet.class, configurationPolicy = ConfigurationPolicy.REQUIRE)
+@SlingServletResourceTypes(
+        resourceTypes = PublishUrlServlet.RESOURCE_TYPE,
+        methods = HttpConstants.METHOD_GET,
+        extensions = PublishUrlServlet.JSON_EXTENSION
+)
+@Designate(ocd = PublishUrlServlet.PublishUrlServletConfig.class)
+public class PublishUrlServlet extends SlingSafeMethodsServlet implements Serializable {
```

Comment:
I was thinking this class could also bind to the Feature OSGi service interface, and implement the Feature contract - so there would be only 1 Java class for this feature. 

That said, Im not sure if there is value decoupling this servlet from the feature.
If not, then we could combine, but I'd suggest renaming this class to be very clearly only inteded for the "AEM Sites  Copy Publish URL" use case.

Also, neither here nor there - but i noticed that it now shows many URLs (that may not Publish URLs; ie. Local, Author, etc.) ... That said, I dont have a better name for it either :)

Reply:
Could be named `ExternalizedUrlsServlet`